model/post.go

```diff
@@ -432,6 +435,38 @@ func (o *Post) ChannelMentions() []string {
 	return ChannelMentions(o.Message)
 }
 
+// DisableMentionHighlights disables a posts mention highlighting and returns the first channel mention that was present in the message.
+func (o *Post) DisableMentionHighlights() string {
+	mention, hasMentions := findAtChannelMention(o.Message)
+	if hasMentions {
+		o.AddProp(POST_PROPS_MENTION_HIGHLIGHT_DISABLED, true)
+	}
+	return mention
+}
+
+// DisableMentionHighlights disables mention highlighting for a post patch if required.
+func (o *PostPatch) DisableMentionHighlights() {
+	_, hasMentions := findAtChannelMention(*o.Message)
+	props := StringInterface{}
+	if o.Props != nil {
+		props = *o.Props
+	}
+
+	if hasMentions && props[POST_PROPS_MENTION_HIGHLIGHT_DISABLED] != true {
```

Comment:
need this?

Reply:
Yea - I refactored this function to be more readable without temporary props 

In hindsight the `props[POST_PROPS_MENTION_HIGHLIGHT_DISABLED] != true` check was overkill since the readability suffered significantly in an attempt to skip a redundant assignment so i removed that part 

I changed the method to just assign Props a pointer to an empty map if its nil to avoid a panic  
and then follow it up with `(*o.Props)[POST_PROPS_MENTION_HIGHLIGHT_DISABLED] = true` with no extra condition.

Reply:
nitpick: If you've refactored this once, might be worth refactoring the preceding DisableMentionHighlights to use the same "if" pattern.